### PR TITLE
[DOCS] Simplify snippet checker logic to catch stray tags in CI

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Simplify docs snippet checker by using `docusaurus` and `grep`

### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: azure-pipelines-docs-integration.yml [hunk 1]
```diff
@@ -73,7 +73,9 @@ stages:
     - job: line_number_checker
       condition: or(eq(stageDependencies.scope_check.changes.outputs['CheckDocsChanges.DocsChanged'], true), eq(variables.isMain, true))
       steps:
-      - bash: python scripts/validate_docs_snippet_line_numbers.py
+      - script: |
+          yarn install
+          python scripts/validate_docs_snippets.py
```
Comment (Shinnnyshinshin) [inline azure-pipelines-docs-integration.yml:78]: 🚀

File: scripts/validate_docs_snippets.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+"""
+Purpose: To ensure that no stray snippet opening/closing tags are present in our production docs
+
+In short, this script creates a temporary Docusaurus build and utilizes grep to parse for stray tags.
```
Comment (Shinnnyshinshin) [inline scripts/validate_docs_snippets.py:4]: Love this

**Review by billdirks — APPROVED** [review]
LGTM.

**Review by Shinnnyshinshin — APPROVED** [review]
Thank you so much for this update. LGTM

**Pull request merged**